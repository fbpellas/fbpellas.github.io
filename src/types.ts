export interface Pronunciation {
  phoneme: string;
  graphemes: string;
  examples: string;
  audioPhoneme: string;
  audioExamples: string;
}

export interface Breadcrumb {
  anchor: string;
  title: string;
}

export enum QuizIndex {
  OddPhonemeOut = 0,
  PhoneticSpelling = 1,
  SameWordsDifferentStress = 2,
  WhereIsTheStress = 3,
  GuessThePattern = 4,
  ShoppingForAPresent = 5
}
