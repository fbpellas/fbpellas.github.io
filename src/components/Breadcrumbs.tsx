import * as React from 'react';
import { generateBreadcrumbs } from '../utils';
import { Breadcrumb } from 'react-bootstrap';
import { QuizIndex } from '../types';

export interface BreadcrumbsProps {
  page: string;
  indexCarousel: QuizIndex;
}

const Breadcrumbs: React.FC<BreadcrumbsProps> = (props) => {
  const { indexCarousel, page } = props;
  const breadcrumbs = generateBreadcrumbs(page, indexCarousel);

  const { length } = breadcrumbs;

  if (length <= 1) return null;

  return (
    <Breadcrumb>
      {breadcrumbs.map((breadcrumb, index) => {
        const { anchor, title } = breadcrumb;
        const isLast = index === length - 1;
        if (isLast) {
          return <Breadcrumb.Item active key={anchor}>{title}</Breadcrumb.Item>;
        }

        return <Breadcrumb.Item href={`#${anchor}`} key={anchor}>{title}</Breadcrumb.Item>;
      })}
    </Breadcrumb >
  );
};

export { Breadcrumbs }