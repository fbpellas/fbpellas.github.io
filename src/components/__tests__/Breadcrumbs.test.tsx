import React from 'react';
import { render } from '@testing-library/react';
import { Breadcrumbs } from '../Breadcrumbs';
import { QuizIndex } from '../../types';

test('Breadcrumbs', () => {
  const indexCarousel = QuizIndex.GuessThePattern;
  const { container, rerender } = render(<Breadcrumbs indexCarousel={indexCarousel} page="home" />);
  expect(container).toBeEmptyDOMElement();

  rerender(<Breadcrumbs indexCarousel={indexCarousel} page="vowels" />);
  expect(container).toMatchSnapshot('some breadcrumbs');
});
