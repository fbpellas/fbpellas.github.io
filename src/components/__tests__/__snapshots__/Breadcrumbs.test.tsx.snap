// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs 1`] = `<div />`;

exports[`Breadcrumbs: some breadcrumbs 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
  >
    <ol
      class="breadcrumb"
    >
      <li
        class="breadcrumb-item"
      >
        <a
          href="#home"
        >
          Home
        </a>
      </li>
      <li
        class="breadcrumb-item"
      >
        <a
          href="#phonemes"
        >
          Phonemes
        </a>
      </li>
      <li
        aria-current="page"
        class="breadcrumb-item active"
      >
        Vowels
      </li>
    </ol>
  </nav>
</div>
`;
